desktop-app/app/shotcut-manager/main-shortcut-manager.js

```diff
@@ -0,0 +1,66 @@
+import { ipcMain, globalShortcut } from 'electron';
+import {
+    REGISTER_CHANNEL,
+    UNREGISTER_CHANNEL,
+    REGISTER_REPLY_CHANNEL,
+    UNREGISTER_REPLY_CHANNEL,
+    GET_ALL_CHANNEL,
+    CLEAR_CHANNEL,
+    ShortcutDefinition,
+    validateDefinition
+} from './shared';
+
+let shortcuts: Map<string, ShortcutDefinition>;
+let mainWindow;
+
+function validate(shortcut: ShortcutDefinition, checkUnique = false) {
+    return validateDefinition(shortcut) && !(checkUnique && shortcuts.has(shortcut.id));
+}
+
+export function registerShortcut(shortcut: ShortcutDefinition): boolean {
+    if (!validate(shortcut, true)) return false;
+    const { id, accelerators } = shortcut;
+    shortcuts.set(id, shortcut);
+    let ok = true;
+    accelerators.forEach(acc => {
+        ok = ok && globalShortcut.register(acc, () => mainWindow.webContents.send(id));
```

Comment:
@jjavierdguezas I think we should not use `globalShortcut` for all the accelerators. 

`globalShortcut` will hijack this short cut at the OS level so it might affect other apps that have similar key combinations registered as a shortcut. We ran into a problem when we used `globalShortcut` for `Ctrl+R` for reload. It was preventing the shortcut functionality in Chrome to work when Responsively was open. 

So we fixed it with the `accelerators` field in the `Menu` object. There should we a way to register these as app-level shortcuts instead of global so that it will be invoked only when the app is in focus.

Reply:
oops, I hadn't thought of that...good catch!
maybe we can use [this](https://github.com/parro-it/electron-localshortcut)